File: src/Features/Str_To_Pinyin.php
<?php

namespace xltxlm\str\Features;

use Overtrue\Pinyin\Pinyin;

/**
 * 中文转拼音;
 */
class Str_To_Pinyin extends Str_To_Pinyin\Str_To_Pinyin_implements
{


    /**
     * Str_To_Pinyin constructor.
     */
    public function __construct(string $Value = '')
    {
        if ($Value) {
            $this->setValue($Value);
        }
    }

    public function __invoke()
    {
        $pinyin = new Pinyin();
        $item = trim(strtr($this->getValue(), ["'" => '']));
        $sentence = $pinyin->sentence($item);
        return strtr(preg_replace('/[ ，,]/', $this->getGlue(), strtoupper($sentence)), ['__' => "_"]);
    }


}

File: go/Features/Str_To_Pinyin.go
package Features

type Str_To_Pinyin struct {
    /* 需要处理的文字 */
    Value string

    /* 拼音中间的链接字符串,默认下划线 */
    Glue string

}

func NewStr_To_Pinyin(Value string,Glue string) *Str_To_Pinyin{
    var this = new(Str_To_Pinyin)
    this.SetValue(Value);
    this.SetGlue(Glue);
    return this
}

func (this *Str_To_Pinyin)GetValue() string{
    return this.Value;
}

func (this *Str_To_Pinyin)SetValue(Value string) *Str_To_Pinyin{
    this.Value = Value;
    return this
}
func (this *Str_To_Pinyin)GetGlue() string{
    return this.Glue;
}

func (this *Str_To_Pinyin)SetGlue(Glue string) *Str_To_Pinyin{
    this.Glue = Glue;
    return this
}

/**
    */
func (this *Str_To_Pinyin)__invoke(){

}

